### ISSUE TITLE
Fatal error: Call to a member function get_finder() on a non-object in /****/****/public_html/forum/includes/functions_module.php on line 1053

### ISSUE THREAD

**Issue by electro143** [fanno/org.jfusion.phpbb31#6]
When I trying to acces my user control panel or my moderator user panel it give me that error : Fatal error: Call to a member function get_finder() on a non-object in /***_/**_*/public_html/forum/includes/functions_module.php on line 1053

**Comment by fanno** [comment]
comittedchange that should fix this one.

**Comment by electro143** [comment]
I've tryed your commit and it working well!I've tweak the css and it begin to look good! I have one major problem thought...When I create a new username,the forum send me an activation link and when I click on it is saying to me that this user account was already activated...But it is not...I try to connect but it is not working...I see the new username in the forum ...But I cant connect with them...

**Comment by electro143** [comment]
OK i think I it working now...But i have to manually sync the users each time there is a new one?

**Comment by fanno** [comment]
What is master/slave ?

And no sync should only need to be run "once" at setup.

**Comment by electro143** [comment]
joomla is master and phpbb is slave
![image](https://cloud.githubusercontent.com/assets/9616210/5076952/041206a4-6e6d-11e4-8b35-1aa9abb3165b.jpg)

**Comment by fanno** [comment]
> When I create a new username,the forum send me an activation link and when I click on it is saying to me that this user account was already activated

The forum should not send an email since registration should happen in joomla? if you register in phpbb that may be part of the problem.

> I try to connect but it is not working...I see the new username in the forum ...But I cant connect with them...

What do you mean ?

-Thanks

**Comment by electro143** [comment]
The "new member" section on buttom of the forum and in my administrator phpbb panel...I see the new members....But I cant connect ...(ive created  a new member for a try).When I try to activate my new account it is saying that the username is already activated...but it is not

**Comment by electro143** [comment]
The only way for new members to work is to sync them manually right now :-/

**Comment by fanno** [comment]
Right, i still don't understand why phpbb would send out a email. may be a phpbb behavior change.

That said i do not understand what you mean by "i cant connect"

If the user is active in joomla, it should be active in phpbb. assuming there is no change to this part of the phpbb database.

The the member you create is that created in joomla ? or ?

**Comment by electro143** [comment]
Ive created a new member using the jfusion login engine...but i a,sure it was a phpbb registration when i try to register ...give it a try you will have a better idea i think...

**Comment by fanno** [comment]
> sure it was a phpbb registration when i try to register

so you mean you are registration throw frameless integration ?

of so this is still not throw joomla. but it would explain why email is sent and why it could cause incorrect behavior if there are global's missing. or something.

**Comment by electro143** [comment]
yes ...the registration part happen all in frameless

**Comment by fanno** [comment]
Then it's properly the cause. it can be hard to track down.

There are a few major problems with integration to phpbb 3.1 and frameless i am not sure if i can solve. =/

Using extensions and url's to /joomla/menu_alias/app.php/demo/world is causing major problems and there may be no way to fix this ..

There are no way that i see to correct the path used in app.php to correctly use the route path /demo/world instead of /joomla/menu_alias/app.php/demo/world.

If this can't be solved we may have to drop frameless support for phpbb =/

This would defiantly mean we it would be logical to create a separate plugin

I am still working on the auth mod replacement for 3.1 but it is quite different, and i might make it so you have to install it manually in to phpbb since it may also have to be manually enabled but we will see.

**Comment by electro143** [comment]
Ok...thank for your help... i will try some tweak in phpbb/joomla myself and contact you if  i found what causing this problem...A+++

**Comment by fanno** [comment]
I will also give it a try. But i won't spend to much time on this unless i can solve the problem with app.php since if i can't fix this. it looks really bleak =/ as this is currently a game stopper for integration

**Comment by electro143** [comment]
I think ive found a really easy fix for that...I just desactivate jfusion login and activated the joomla login...I had no problem after that creating a new account with a email activation link on email (joomla style registration)...It also created the same name in phpbb!Everything seem to work nicely now...The only "issue" remaining is when Im going in phpbb it seem to disconnect from joomla...but it stay connected in phpbb...

**Comment by electro143** [comment]
Rectification : it seem to "disconnect "from joomla when i click on a section of the forum...

**Comment by fanno** [comment]
as in login session lost or the virtual integration is lost or ....  ?

**Comment by electro143** [comment]
The integretion in joomla is lost but not in phpbb(it is staying connected here)...Seem to be an intrmittent problem thought...

I finall6 use the jfusion login module but changed the parameter of login to joomla...seem to work well except for the disconnecting part and cant get avatar from phpbb to show in the login module...

**Comment by fanno** [comment]
Perhaps phpbb 3.1 has different avatar system i have not looked at that.

As to staying connected, it may be one of two things (or something new)
1. phpbb plugin has two login methods, changing it may remove the problem.
2. It may be you need to use the session plugin (session sync).

**Comment by electro143** [comment]
In my php plugin i have two type of login :Direct and indirect...It was in ''indirect mode'' I tried to put it to direct mode,but when im trying to connect it give me that error:

Fatal error: Call to a member function header() on a non-object in /***_/**_*/public_html/forum/phpbb/session.php on line 223

**Comment by fanno** [comment]
Thanks =) i will add the global when i get home.

**Comment by electro143** [comment]
No problem!I think i will going throught all frameless problem integration with you hahaha...Great support! :-D

**Comment by electro143** [comment]
Additional note: The ''disconnecting'' problem from joomla seem to only happen with my administrator account...I've tried it with a created account and it seem to stay connected

**Comment by fanno** [comment]
add globals

**Comment by electro143** [comment]
Dont now if this issue is due to this error but when I login logout of joomla,im losing my phpbb administrator and modo privilege...When i logout directly from phpbb i doesnt lose my privileges...Ill try your new global in the next hours and see if it help...thank

**Comment by electro143** [comment]
Ive tried the direct connect and it no longer indicate an error...But I did not corrected my last bug...I still lost my privilige when I disconnect from joomla...Ive tried nomitade a standard user modo...It worked well...I disconnected from phpbb and it worked well too...But when I disconnected from joomla ive lost my privilege...

**Comment by fanno** [comment]
user group configuration is properly incorrect.

**Comment by electro143** [comment]
![untitled-2](https://cloud.githubusercontent.com/assets/9616210/5099440/1a489b84-6f64-11e4-8add-0a903758e934.gif)
What is wrong?

**Comment by fanno** [comment]
hmm if only default group is configured that should work. unless group configuration in php 3.1 has changed.  or perhaps phpbb 3.1 do not line administrator group in secondary group.

**Comment by electro143** [comment]
It is now working I just unchecked "mettre a jour lors de changement"

**Comment by fanno** [comment]
must be something with user group update then,
